File: examples/gcd.py
#
# jython examples for jas.
# $Id$
#

from java.lang import System

from jas import Ring, PolyRing, QQ, ZZ, GF
from jas import terminate, startLog

# polynomial examples: gcd

#r = Ring( "Mod 1152921504606846883 (x,y,z) L" );
#r = Ring( "Rat(x,y,z) L" );
#r = Ring( "C(x,y,z) L" );
#r = Ring( "Z(x,y,z) L" );
r = PolyRing( QQ(), "x,y,z", PolyRing.lex );
print "Ring: " + str(r);
print;

#[one,x,y,z] = r.gens();

a = r.random();
b = r.random();
c = abs(r.random());
#c = 1; 
#a = 0;


print "a = ", a;
print "b = ", b;
print "c = ", c;
print;

ac = a * c;
bc = b * c;

print "ac = ", ac;
print "bc = ", bc;
print;

t = System.currentTimeMillis();
d = r.gcd(ac,bc);
t = System.currentTimeMillis() - t;

#d = d.monic();
print "d = " + str(d);
print "d = " + str(d.monic());

m = c % d;
## print "m = ", m;
## print;

if m.isZERO():
    print "gcd time =", t, "milliseconds,", "isGcd(c,d): true" ;
else:
    print "gcd time =", t, "milliseconds,", "isGcd(c,d): ",  str(m);
print;


#startLog();
terminate();

File: examples/gcd.rb
#
# jruby examples for jas.
# $Id$
#

require "examples/jas"

# polynomial examples: gcd

#r = PolyRing.new( ZM(1152921504606846883), "(x,y,z)", PolyRing.lex );
r = PolyRing.new( QQ(), "(x,y,z)", PolyRing.lex );
#r = PolyRing.new( CC(), "(x,y,z)", PolyRing.lex );
#r = PolyRing.new( ZZ(), "(x,y,z)", PolyRing.lex );

puts "Ring: " + str(r);
puts;

one,x,y,z = r.gens();

a = r.random();
b = r.random();
c = r.random().abs();
#c = 1; 
#a = 0;

puts "a = " + str(a);
puts "b = " + str(b);
puts "c = " + str(c);
puts;

ac = a * c;
bc = b * c;

puts "ac = " + str(ac);
puts "bc = " + str(bc);
puts;

#startLog();

t = System.currentTimeMillis();
d = r.gcd(ac,bc);
t = System.currentTimeMillis() - t;

#d = d.monic();
puts "d = " + str(d);
puts "d = " + str(d.monic());
puts;

puts "gcd time = " + str(t) + " milliseconds" ;

if r.ring.coFac.isField()
   m = c % d;
   #puts "m = " + str(m);
   puts;
   if m.isZERO()
      puts "isGcd(c,d): true" ;
   else
      puts "isGcd(c,d): " + str(m);
   end
end
puts;


#startLog();
terminate();
